integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java

```diff
@@ -20,37 +21,41 @@
 
 public class BridgeSteps {
 
-    @Given("get list of Bridge instances returns HTTP response code (\\d+)$")
-    public void authenticationIsEnabled(int responseCode) {
-        BridgeResource.getBridgeListFails(responseCode);
+    private TestContext context;
+
+    public BridgeSteps(TestContext context) {
+        this.context = context;
     }
 
-    @Given("get list of Bridge instances with access token doesn't contain randomly generated Bridge")
-    public void generateRandomBridgeName() {
-        StepsContext.randomBridgeName = "bridge-" + UUID.randomUUID().toString().substring(0, 4);
-        BridgeListResponse response = BridgeResource.getBridgeList(BridgeUtils.retrieveAccessToken());
-        assertThat(response.getKind()).isEqualTo("BridgeList");
-        assertThat(response.getItems()).noneMatch(b -> b.getName().equals(StepsContext.randomBridgeName));
+    @Given("get list of Bridge instances returns HTTP response code (\\d+)$")
+    public void authenticationIsEnabled(int responseCode) {
+        BridgeResource.getBridgeListFails(context.getManagerToken(), responseCode);
     }
 
     @When("create a Bridge with randomly generated name with access token")
     public void createRandomBridge() {
-        BridgeResponse response = BridgeResource.addBridge(BridgeUtils.retrieveAccessToken(),
-                StepsContext.randomBridgeName);
-        assertThat(response.getName()).isEqualTo(StepsContext.randomBridgeName);
+        String bridgeName = Utils.generateId("bridge");
+        while (isBridgeExisting(bridgeName)) {
```

Comment:
In theory this can end in endless loop. Would be good to have limited tries to be sure.

Reply:
see https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/367/commits/4e60f40dde1a67e05e60051b509c961c244389f6